### PR TITLE
Documentation improvements.

### PR DESCRIPTION
**Pull request by apdavison**

### CODE REVIEW

**Comment by apdavison** [comment]
I'm not going to fix the remaining PEP8 issues (since the PyNN IOs will be removed in v0.7...)

This is ready to merge.

File: doc/source/io.rst [hunk 1]
```diff
@@ -114,12 +114,13 @@ Lazy option
 
 In some cases you may not want to load everything in memory because it could be too big.
 For this scenario, some IOs implement ``lazy=True/False``. With ``lazy=True`` all arrays will have a size of zero,
-but all the metadata will be loaded. lazy_shape attribute is added to all object that inherit Quantitities
-or numpy.ndarray (AnalogSignal, AnalogSignalArray, SpikeTrain)  and to object that have array like attributes (EpochArray, EventArray)
-In that cases, lazy_shape is a tuple that have the same shape with lazy=False. To known if a class support this lazy mode use ``ClassIO.support_lazy``
+but all the metadata will be loaded. The *lazy_shape* attribute is added to all array-like objects
+(AnalogSignal, AnalogSignalArray, SpikeTrain, Epoch, Event).
```
Comment (samuelgarcia) [inline doc/source/io.rst]: AnalogSignalArray do not exists any more. I think you mean IrregularySampledSignal

File: doc/source/io_developers_guide.rst [hunk 2]
```diff
@@ -49,21 +48,21 @@ Tests
...
+The test scripts download all files from `gin-gnode`_ and stores them locally in ``/tmp/files_for_tests/``.
 Subsequent test runs use the previously downloaded files, rather than trying to download them each time.
 
-Each test must have at least one class that inherits ``BaseTestRawIO`` and that have 3 attributes:
-  * rawioclass: the class
-  * entities_to_test: a list of file (or directoris) to be tested one by one
-  * files_to_download: a list of file to download (sometime biger than entities_to_test)
+Each test must have at least one class that inherits ``BaseTestRawIO`` and that has 3 attributes:
+  * ``rawioclass``: the class
+  * ``entities_to_test``: a list of files (or directoris) to be tested one by one
```
Comment (samuelgarcia) [inline doc/source/io_developers_guide.rst]: directories

**Comment by samuelgarcia** [comment]
Sorry so having made the doc so drafty.
Thank you very much for the rewrite!!

**Comment by samuelgarcia** [comment]
Can I merge ?

File: doc/source/developers_guide.rst [hunk 3]
```diff
@@ -38,11 +38,9 @@ a GitHub account and then set to watch the repository at `GitHub Repository`_
 Requirements
 ------------
 
-    * Python_ 2.6, 2.7, 3.3-3.5
+    * Python_ 2.7, 3.4 or later
     * numpy_ >= 1.7.1
     * quantities_ >= 0.9.0
```
Comment (samuelgarcia) [inline doc/source/developers_guide.rst]: I have lots of errors with nose test old version of wuantittes in an old env.
So we should force latest version here 0.12.1 

this is due to recent change in spiketrains:
```
    @property
    def times(self):
        return pq.Quantity(self)
```

Comment (apdavison) [reply doc/source/developers_guide.rst]: agreed, but let's update the requirements (which appear in several places) in a separate pull request

**Comment by samuelgarcia** [comment]
OK. I let make the other PR for requirement.

**Pull request merged**